ouroboros-network/src/Ouroboros/Network/BlockFetch/State.hs

```diff
@@ -67,19 +75,24 @@ fetchLogicIterations decisionTracer clientStateTracer
                      fetchTriggerVariables
                      fetchNonTriggerVariables =
 
-    iterateForever initialFetchStateFingerprint $ \stateFingerprint ->
+    iterateForever initialFetchStateFingerprint $ \stateFingerprint -> do
 
       -- Run a single iteration of the fetch logic:
       --
       -- + wait for the state to change and make decisions for the new state
       -- + act on those decisions
-
-      fetchLogicIteration
+      start <- getMonotonicTime
+      stateFingerprint' <- fetchLogicIteration
         decisionTracer clientStateTracer
         fetchDecisionPolicy
         fetchTriggerVariables
         fetchNonTriggerVariables
         stateFingerprint
+      end <- getMonotonicTime
+      let delta = diffTime end start
+      -- Limit descision making to once every decisionLoopInterval.
+      when (delta < decisionLoopInterval fetchDecisionPolicy) $ threadDelay delta
```

Comment:
As you mentioned on our Slack thread: `threadDelay (decisionLoopInterval fetchDecisionPolicy - delta)`.